frontend/components/FieldColor/FieldColorEdit.jsx

```diff
@@ -0,0 +1,262 @@
+     */
+    group: proptypes.string,
+
+    /**
+     * The name of the field within the collection. May be a path using
+     * dot-notation.
+     */
+    name: proptypes.string,
+
+    /**
+     * A callback to be fired whenever the field wants to update its value to
+     * a successful state. The function receives the name of the field and the
+     * new value as arguments.
+     */
+    onChange: proptypes.string,
+
+    /**
+     * A callback to be fired whenever the field wants to update its value to
+     * or from an error state. The function receives the name of the field, a
+     * Boolean value indicating whether or not there's an error and finally the
+     * new value of the field.
+     */
+    onError: proptypes.string,
+
+    /**
+     * Whether the field is required.
+     */
+    required: proptypes.bool,
+
+    /**
+     * The field schema.
+     */
+    schema: proptypes.object,
+
+    /**
+     * The field value.
+     */
+    value: proptypes.bool
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state.pickerVisible = false
+
+    this.picker = false
+    this.pickerEventHandler = this.handlePickerClick.bind(this, true)
+    this.pickerOutsideEventHandler = this.handlePickerClick.bind(this, false)
+    this.losingFocusTimeout = null
+    this.state.hasFocus = false
+  }
+
+  render() {
+    let {
+      comment,
+      config,
+      displayName,
+      error, 
+      name,
+      required,
+      schema, 
+      value
+    } = this.props
+    const {hasFocus} = this.state
+    const {pickerVisible} = this.state
+    const publishBlock = schema.publish || {}
+
+    return (
+      <Label
+        className={styles.label}
+        error={error}
+        errorMessage={typeof error === 'string' ? error : null}
+        hasFocus={hasFocus}
+        label={displayName}
+        comment={comment || (required && 'Required') || (publishBlock.readonly && 'Read only')}
+      > 
+        <TextInput
+          name={name}
+          onBlur={this.handleFocus.bind(this, false)}
+          onChange={this.handleChange.bind(this)}
+          onFocus={this.handleFocus.bind(this, true)}
+          readonly={publishBlock.readonly === true}
+          type="text"
+          value={value}
+        />
+
+        <div 
+          class={styles.swatch}
+          style={value ? `background-color:#${value}` : ''}
+        />
+
+        {pickerVisible &&
+          <div ref={this.handlePickerRef.bind(this)}>
+            <ColorPicker
+              className={styles.picker}
+              color={value}
+              onChange={this.handlePickerChange.bind(this)}
+              format={publishBlock.format || 'hex'}
+            />
+          </div>
+        }
+      </Label>
+    )
+  }
+
+  componentDidMount() {
+    window.addEventListener('click', this.pickerOutsideEventHandler)
+  }
+
+  componentWillUnmount() {
+    if (this.picker) {
+      this.picker.removeEventListener('click', this.pickerOutsideEventHandler)
```

Comment:
The `pickerOutsideEventHandler` handler was attached to `window`. Shouldn't this be `window.removeEventListener`?